src/tests/multihost/alltests/conftest.py

```diff
@@ -111,6 +111,17 @@ def removesssd():
     return _modifysssd
 
 
+@pytest.fixture(scope='function')
+def capture_sssd_logs(session_multihost, request):
+    """This will print sssd logs in case of test failure"""
+    yield
+    if request.session.testsfailed:
+        client = session_multihost.client[0]
+        print(f"\n\n===Logs for {request.node.name}===\n\n")
+        for data_d in client.run_command("ls /var/log/sssd/").stdout_text.split():
```

Comment:
Instead of the loop just use bash wildcard "cat /var/log/sssd/*" or at least make sure to print each filename as well so we can see what log it is.

Reply:
I tried both  but did not worked, this was the last option i get working

Reply:
client.run_command(f'echo "--- {data_d} ---"; cat /var/log/sssd/{data_d}')

Reply:
2023-01-06 12:34:27,087 - sssd.testlib.common.qe_class.QeHost.client1.cmd3 - DEBUG - sssd_pam.log
2023-01-06 12:34:27,087 - sssd.testlib.common.qe_class.QeHost.client1.cmd3 - DEBUG - Exit code: 0
2023-01-06 12:34:27,129 - sssd.testlib.common.qe_class.QeHost.client1.ParamikoTransport - INFO - RUN echo "--- sssd_example1.log ---"; cat /var/log/sssd/sssd_example1.log
2023-01-06 12:34:27,129 - sssd.testlib.common.qe_class.QeHost.client1.cmd4 - DEBUG - RUN echo "--- sssd_example1.log ---"; cat /var/log/sssd/sssd_example1.log
2023-01-06 12:34:27,163 - sssd.testlib.common.qe_class.QeHost.client1.cmd4 - DEBUG - -bash: line 1: cd: /root/multihost_tests: No such file or directory
2023-01-06 12:34:27,163 - sssd.testlib.common.qe_class.QeHost.client1.cmd4 - DEBUG - -bash: line 2: /root/multihost_tests/env.sh: No such file or directory
2023-01-06 12:34:27,164 - sssd.testlib.common.qe_class.QeHost.client1.cmd4 - DEBUG - --- sssd_example1.log ---
2023-01-06 12:34:27,165 - sssd.testlib.common.qe_class.QeHost.client1.cmd4 - DEBUG - (2023-01-05  3:33:04): [be[example1]] [server_setup] (0x3f7c0): Starting with debug level = 0x2f7f0
2023-01-06 12:34:27,165 - sssd.testlib.common.qe_class.QeHost.client1.cmd4 - DEBUG - (2023-01-05  3:33:04): [be[example1]] [server_setup] (0x0400): CONFDB: /var/lib/sss/db/config.ldb
2023-01-06 12:34:27,165 - sssd.testlib.common.qe_class.QeHost.client1.cmd4 - DEBUG - (2023-01-05  3:33:04): [be[example1]] [dp_get_options] (0x0400): Option lookup_family_order has value ipv4_first
2023-01-06 12:34:27,165 - sssd.testlib.common.qe_class.QeHost.client1.cmd4 - DEBUG - (2023-01-05  3:33:04): [be[example1]] [dp_get_options] (0x0400): Option dns_resolver_timeout has

Reply:
I tried all the option you mentioned above before pushing this PR. Non works . this loop option is only way to do the work correctly and it also prints log names only once

Reply:
ok last  one works , added